### PR TITLE
Alternate approach to fix tilde expansion

### PR DESCRIPTION
**Pull request by xdg**
This is alternative solution to #191 to address #190.

### CODE REVIEW

**Comment by xdg** [comment]
@ap, please take a look at this alternative.  I'm leery of trying to regex grab the tilde before canonicalization, so I'd rather just flag it early and pull it off later.  I kept your tests and added some more.

**Comment by haarg** [comment]
This looks reasonable to me.

**Comment by ap** [comment]
LGTM also.

File: lib/Path/Tiny.pm [hunk 1]
```diff
@@ -247,11 +250,17 @@ sub path {
     }
 
     # do any tilde expansions
-    if ( $path =~ m{^(~[^/]*).*} ) {
+    if ($starts_with_tilde) {
         require File::Glob;
-        my ($homedir) = File::Glob::bsd_glob($1);
+        my ($tilde_part) = $path =~ m{^(~[^/]*)};
```
Comment (ap) [inline lib/Path/Tiny.pm]: Hmm. Is there any scenario where this match might fail? The rest of the block assumes it never will.

**Comment by ap** [comment]
So… maybe like this?

````perl
if ( $path =~ m{^(~[^/]*)} ) {
    if ($do_tilde_expansion) {
        require File::Glob;
        my ($homedir) = File::Glob::bsd_glob($1);
        $homedir =~ tr[\\][/] if IS_WIN32();
        $path =~ s{^(~[^/]*)}{$homedir};
    }
    else {
        # didn't originally start with tilde, but now does after
        # canonicalization, so we 'escape' it via prepending './'
        $path = "./$path";
    }
}
````

Or even

````perl
if ( $path =~ m{^(~[^/]*)}g ) {
    if ($do_tilde_expansion) {
        require File::Glob;
        my ($homedir) = File::Glob::bsd_glob($1);
        $homedir =~ tr[\\][/] if IS_WIN32();
        substr $path, 0, pos $path, $homedir;
    }
    else {
        # didn't originally start with tilde, but now does after
        # canonicalization, so we 'escape' it via prepending './'
        $path = "./$path";
    }
}
````

**Comment by ap** [comment]
@xdg: ping?

**Comment by xdg** [comment]
I don't have tuits right now but will get to it by or worst-case during TPC.

**Comment by ap** [comment]
@xdg: ping?

**Comment by ap** [comment]
@xdg: ping?

**Comment by xdg** [comment]
Closing.

**Comment by ap** [comment]
But… why?

**Pull request closed without merging**